x/blockdb/database.go

```diff
@@ -223,6 +225,7 @@ func New(config DatabaseConfig, log logging.Logger) (*Database, error) {
 				f.Close()
 			}
 		}),
+		entryCache: lru.NewCache[BlockHeight, BlockData](config.EntryCacheSize),
```

Comment:
this is one approach of doing it.

I think another perhaps more elegant approach would be to have a wrapper that intercepts Put and Has and Get and has the cache. The latter is much more testable as you can mock the inner backend. 

I'm fine with this approach though, just wanted to suggest another idea.

Reply:
I get the duplication and I still think it's worth it, but I'll defer to your decision 🤷